libmachine/mcnutils/crc.go

```diff
@@ -229,8 +231,21 @@ func (b *B2dUtils) CopyDiskToMachineDir(diskURL, machineName string) error {
 
 	// By default just copy the existing "cached" iso to the machine's directory...
 	if diskURL == "" {
-		log.Infof("Copying %s to %s...", b.path(), machineIsoPath)
+		log.Infof("\n\nCopying %s to %s...\n\n\n", b.path(), machineIsoPath)
 		return CopyFile(b.path(), machineIsoPath)
+	} else if strings.HasPrefix(diskURL, "file:") {
+		// Local disk image so we can avoid copying by making use of backing image feature of qcow2
+		components := strings.Split(diskURL, "://")
+		if len(components) < 2 {
+			return fmt.Errorf("Invalid URL: %s", diskURL)
+		}
+		diskPath := components[1]
+		log.Infof("Create disk image at %s (backing store: %s)", machineIsoPath, diskPath)
+		command := exec.Command("qemu-img", "create", "-b", diskPath, "-f", "qcow2", machineIsoPath)
+		output, err := command.CombinedOutput()
+		log.Debugf("Output from `%s` command:\n%s", command.String(), output)
+
+		return err
```

Comment:
@zeenix This `CopyDiskToMachineDir` is used by all other driver not only for libvirt so adding change here is no go, also now user have to make sure `qemu-img` command present on the system, I think this whole logic should go on libvirt driver side not here.

Reply:
You're right. I'll address individual review comments here and then close the PR here, move the issue back and then create a PR on `machine-driver-libvirt`.